src/averageAges.js

```diff
@@ -55,7 +67,21 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(person => people.some(child => child.name === person.mother)
+      && person.sex === 'm')
+    : people.filter(person => people.some(child =>
+      child.name === person.mother));
+
+  const difference = children.map(child => {
+    const motherChild = people.find(mother => mother.name === child.mother);
+
+    return child.born - motherChild.born;
+  });
+
+  const age = difference.reduce((sum, x) => (sum + x)) / difference.length;
```

Comment:
you use identical reduce three times, it is better to create a separate function with this reduce and use this function